### ISSUE TITLE
More features for Observability Analysis

### ISSUE THREAD

**Issue by alex-w** [Stellarium/stellarium#464]
_Original report by Keatah:_ https://bugs.launchpad.net/stellarium/+bug/1045297

Observability Analysis plugin (ideas for more features)

Things to add to the plugin:
1- Distance from Sun for Solar System objects
2- Heliocentric x,y,z distances
3- More info for phases of the moon - Age, Angular Size..
4- Some sort of graphical popup moon calendar
5- Degree of subtention
6- Some sort of general astronomy calculator (this could be another plug-in or a totally different application entirely)

**Comment by gzotti** [comment]
Some of the numbers should be given only when accuracy issues have been solved. Else people would start using Stellarium as reference solution...

**Comment by axd1967** [comment]
Maybe overkill, but a dialog could be shown that warns the user about the accuracy of the graphs, or some other way to indicate that accuracy is not good for reference use?
Just freewheeling: how about making the lines thicker, as fine lines imply high accuracy?

**Comment by gzotti** [comment]
@axd1967 Do you want to confirm a dialog for every numerical result that is technically wrong in the last displayed decimal figure? You will hurt your index finger from clicking.

Appendix F in the Stellarium User Guide is where you find accuracy estimates. As long as the version number is 0.*, you should not use Stellarium as reference source for numerical ephemerides.

**Comment by axd1967** [comment]
No... I was thinking about a one-time dialog with a checkbox that allows the user to hide it "forever" (or until reset by some other means). The dialog could contain one or more hyperlinks to one of these:

-  https://github.com/Stellarium/stellarium/blob/master/guide/app_accuracy.tex, 
- https://github.com/Stellarium/stellarium/blob/1dbadaec8e50a1cfdc681cfa99f8bc3dbf79414e/guide/app_accuracy.tex#L10-L30
- https://github.com/Stellarium/stellarium/releases/download/v0.19.2/stellarium_user_guide-0.19.2-1.pdf.

**Comment by gzotti** [comment]
I made such a one-time warning for users with too old computers that they should expect graphic problems. Those still reported about graphics problems. "Ah yes, there was some warning, but I clicked continue". Do you think anybody cares for what they confirmed when accepting some popup dialog that you see once just after installation or at first run?

**Comment by alex-w** [comment]
Couple features are implemented in main code already:
* [x] Distance from Sun for Solar System objects
* [x] More info for phases of the moon - Age, Angular Size..

**Comment by alex-w** [comment]
What is "Degree of subtention" in astronomical application?

**Comment by alex-w** [comment]
I mark last item as solved due AstroCalc tool

**Comment by MaraJadeLives** [comment]
As a person doing radio astronomy, would it make sense to have a setting where I could set it, so it doesn't matter if it's night or day when the object is up, just that it's at least a user defined number of degrees away from the sun, for calculating observability?

**Comment by gzotti** [comment]
The Observability plugin is specifically designed for visual observation. It is in need of a thorough rewrite, though. But the AstroCalc is the more likely platform for extensions to select objects by elongation/angular distance.
